contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,507 @@
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            _addScheduleFromTemplate(_beneficiaries[i], _index, _startTime);
+        }
+    }
+
+    /**
+     * @notice Used to bulk revoke vesting schedules for each of beneficiaries
+     * @param _beneficiaries array of beneficiary's addresses
+     */
+    function revokeSchedulesMulti(address[] _beneficiaries) external withPerm(ADMIN) {
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            revokeSchedules(_beneficiaries[i]);
+        }
+    }
+
+    /**
+     * @notice Used to bulk modify vesting schedules for each of beneficiaries
+     * @param _beneficiaries array of beneficiary's addresses
+     * @param _indexes array of beneficiary's indexes of schedule
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     * @param _startTime vesting start time
+     */
+    function modifyScheduleMulti(
+        address[] _beneficiaries,
+        uint256[] _indexes,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime
+    )
+        external
+        withPerm(ADMIN)
+    {
+        require(_beneficiaries.length == _indexes.length, "Arrays sizes mismatch");
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            modifySchedule(_beneficiaries[i], _indexes[i], _numberOfTokens, _duration, _frequency, _startTime);
+        }
+    }
+
+    function _validateSchedule(
+        address _beneficiary,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime
+    )
+        internal
+        view
+    {
+        require(_beneficiary != address(0), "Invalid address");
+        _validateTemplate(_numberOfTokens, _duration, _frequency);
+        require(now < _startTime, "Date in the past");
+    }
+
+    function _validateTemplate(uint256 _numberOfTokens, uint256 _duration, uint256 _frequency) internal pure {
+        require(_numberOfTokens > 0, "Zero amount");
+        require(_duration % _frequency == 0, "Duration and frequency mismatch");
+        uint256 periodCount = _duration.div(_frequency);
+        require(_numberOfTokens % periodCount == 0, "Tokens and periods mismatch");
+    }
+
+    function _sendTokens(address _beneficiary) internal {
+        uint256 amount = _getAvailableTokens(_beneficiary);
+        if (amount > 0) {
+            for (uint256 i = 0; i < schedules[_beneficiary].length; i++) {
+                schedules[_beneficiary][i].availableTokens = 0;
+            }
+            ISecurityToken(securityToken).transfer(_beneficiary, amount);
+            emit SendTokens(_beneficiary, amount);
+        }
+    }
+
+    /**
+     * @notice manually triggers update outside for all schedules (can be used to reduce user gas costs)
+     */
+    function updateAll() external withPerm(ADMIN) {
```

Comment:
As far my understanding `updateAll()` will be used to calculate the `availableTokens` for the beneficiaries. but It will make the execution cost more than it saved.  Suppose there are 5 users and every users period completion within the difference of 1 day (i.e U1 period completion on t1 then others will be t1 + 1, t1 + 2, t1 + 3 ...). then to make available tokens updated you will need to call the `updateAll()` function every day. that lead more cost burning instead of saving. I will recommend calculating the `available tokens` in the fly (at the execution time). you can check other modules implementation (BlackListTransferManager) we are doing this.

If you have any queries please let me know we can chat over slack or do a quick call.